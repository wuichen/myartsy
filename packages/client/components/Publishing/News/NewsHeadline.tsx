import { Box, Flex, Serif } from "@artsy/palette"
import React from "react"

interface NewsHeadlineProps {
  article: any
  editTitle?: any
}

export const NewsHeadline: React.SFC<NewsHeadlineProps> = props => {
  const { article, editTitle } = props
  const { title } = article

  return (
    <Box>
      <Flex flexDirection="column" maxWidth="780" mx="auto" mt="1" mb="3">
        <Serif
          size={["6", "8"]}
          lineHeight={["1.1", "initial"]}
          weight="semibold"
          element="h1"
        >
          {editTitle ? editTitle : title}
        </Serif>
      </Flex>
    </Box>
  )
}
