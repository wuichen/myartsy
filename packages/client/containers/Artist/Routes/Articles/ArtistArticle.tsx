import { Col, Row } from "@artsy/palette"
import React, { SFC } from "react"
import { Media } from "Utils/Responsive"

import {
  Box,
  Flex,
  Image,
  Sans,
  Separator,
  Serif,
  Spacer,
} from "@artsy/palette"

interface ArticleItemProps {
  imageUrl: string
  date: string
  author: string
  title: string
  href: string
  lastChild: boolean
}

export const ArticleItem: SFC<ArticleItemProps> = props => {
  return (
    <>
      <Media at="xs">
        <SmallArticleItem {...props} />
      </Media>
      <Media greaterThan="xs">
        <LargeArticleItem {...props} />
      </Media>
    </>
  )
}

const LargeArticleItem: SFC<ArticleItemProps> = props => {
  const { author, date, href, imageUrl, title, lastChild } = props

  return (
    <>
      <Row
        style={{ cursor: "pointer" }}
        onClick={() => {
          window.location.href = href
        }}
      >
        <Col sm={2}>
          <Date size="3" weight="medium">
            {date}
          </Date>
        </Col>
        <Col sm={8}>
          <Flex>
            <Box pr={12}>
              <Title size="5">
                <a href={href} className="noUnderline">
                  {title}
                </a>
              </Title>
              <Credit size="2" color="black60">
                {author}
              </Credit>
            </Box>
          </Flex>
        </Col>
        <Col sm={2}>
          <a href={href}>
            <Image
              style={{
                height: "100px",
                width: "auto",
                maxWidth: 140,
                maxHeight: 280,
                objectFit: "fill",
              }}
              src={imageUrl}
              mb={2}
            />
          </a>
        </Col>
      </Row>
      {/* FIXME: Weird block height issue... */}
      {!lastChild && <Separator />}
      <Spacer mt={3} />
    </>
  )
}

const SmallArticleItem: SFC<ArticleItemProps> = props => {
  const { author, date, href, imageUrl, title, lastChild } = props

  return (
    <>
      <Flex
        justifyContent="space-between"
        style={{ cursor: "pointer" }}
        onClick={() => {
          window.location.href = href
        }}
      >
        <Box pr={2}>
          <Date size="1" weight="medium">
            {date}
          </Date>
          <Spacer mb={0.5} />
          <Title size="2">
            <a href={href} className="noUnderline">
              {title}
            </a>
          </Title>
          <Spacer mb={0.5} />
          <Credit size="1" color="black60">
            {author}
          </Credit>
        </Box>
        <Image width="70px" height="100%" src={imageUrl} />
      </Flex>
      {/* FIXME: Weird block height issue... */}
      <Spacer mt={2} />
      {!lastChild && <Separator />}
      <Spacer mt={3} />
    </>
  )
}

const Date = Sans
const Title = Serif
const Credit = Serif
