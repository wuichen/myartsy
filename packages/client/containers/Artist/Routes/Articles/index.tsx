import { Articles_artist } from "__generated__/Articles_artist.graphql"
import React from "react"
import { createFragmentContainer, graphql } from "react-relay"
import { ArtistArticlesRefetchContainer as Articles } from "./ArtistArticles"

export interface ArticlesRouteProps {
  artist: Articles_artist
}

export const ArticlesRoute: React.SFC<ArticlesRouteProps> = props => {
  return <Articles artist={props.artist} />
}

export const ArticlesRouteFragmentContainer = createFragmentContainer(
  ArticlesRoute,
  {
    artist: graphql`
      fragment Articles_artist on Artist {
        ...ArtistArticles_artist
      }
    `,
  }
)
