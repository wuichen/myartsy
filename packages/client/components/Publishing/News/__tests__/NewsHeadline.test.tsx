import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"

import { NewsArticle } from "Components/Publishing/Fixtures/Articles"
import { EditableChild } from "Components/Publishing/Fixtures/Helpers"
import { NewsHeadline } from "../NewsHeadline"

it("renders a news headline properly", () => {
  const component = renderer
    .create(<NewsHeadline article={NewsArticle} />)
    .toJSON()
  expect(component).toMatchSnapshot()
})

it("renders a headline with children properly", () => {
  const component = renderer
    .create(
      <NewsHeadline article={NewsHeadline} editTitle={EditableChild("title")} />
    )
    .toJSON()
  expect(component).toMatchSnapshot()
})
