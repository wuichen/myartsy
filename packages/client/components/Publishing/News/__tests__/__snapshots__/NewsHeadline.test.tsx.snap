// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a headline with children properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 780;
  margin-bottom: 16px;
  margin-top: 4px;
  margin-left: auto;
  margin-right: auto;
}

.c1 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-weight: 600;
  font-size: 26px;
  line-height: 1.1;
}

@media screen and (min-width:40em) {
  .c1 {
    font-size: 32px;
  }
}

@media screen and (min-width:40em) {
  .c1 {
    line-height: initial;
  }
}

<div
  className=""
>
  <div
    className="c0"
  >
    <h1
      className="c1"
      fontFamily={
        Object {
          "fontFamily": "'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif",
          "fontWeight": 600,
        }
      }
      fontSize={
        Array [
          "26px",
          "32px",
        ]
      }
    >
      <div>
        Child 
        title
      </div>
    </h1>
  </div>
</div>
`;

exports[`renders a news headline properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 780;
  margin-bottom: 16px;
  margin-top: 4px;
  margin-left: auto;
  margin-right: auto;
}

.c1 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-weight: 600;
  font-size: 26px;
  line-height: 1.1;
}

@media screen and (min-width:40em) {
  .c1 {
    font-size: 32px;
  }
}

@media screen and (min-width:40em) {
  .c1 {
    line-height: initial;
  }
}

<div
  className=""
>
  <div
    className="c0"
  >
    <h1
      className="c1"
      fontFamily={
        Object {
          "fontFamily": "'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif",
          "fontWeight": 600,
        }
      }
      fontSize={
        Array [
          "26px",
          "32px",
        ]
      }
    >
      The oldest known depiction of a supernova was found in a work of 5,000 year old rock art, scientists believe.
    </h1>
  </div>
</div>
`;
